src/pages/index.js

```diff
@@ -1,80 +1,165 @@
+import Card from "../components/Card.js";
 import Section from "../components/Section.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import { api } from "../components/Api.js";
 
 import {
-  initialCards,
   validationObj,
   editButton,
   addButton,
+  profileImage,
+  profilePicForm,
   editForm,
   addCardForm,
   modalImage,
   modalTitle,
 } from "../utils/constants.js";
 
-const cardPopup = new PopupWithForm("#add-card-modal", renderNewCard);
-const profilePopup = new PopupWithForm("#edit-profile-modal", editProfile);
-const locationPopup = new PopupWithImage("#location-modal");
+const popups = {
+  cardPopup: new PopupWithForm("#add-card-modal", renderNewCard),
+  profilePopup: new PopupWithForm("#edit-profile-modal", editProfile),
+  deleteCardPopup: new PopupWithForm("#delete-card-modal", deleteCard),
+  profilePicPopup: new PopupWithForm("#profile-pic-modal", editProfilePic),
+  locationPopup: new PopupWithImage("#location-modal"),
+};
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__name",
-  descriptionSelector: ".profile__subtitle",
-});
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  ".locations"
+const userInfo = new UserInfo(
+  {
+    nameSelector: ".profile__name",
+    descriptionSelector: ".profile__subtitle",
+  },
+  profileImage
 );
+const cardSection = new Section({ renderer: createCard }, ".locations");
 
 // validation class
-const editFormValidation = new FormValidator(validationObj, editForm);
-const cardFormValidation = new FormValidator(validationObj, addCardForm);
+const formValidators = {
+  editFormValidation: new FormValidator(validationObj, editForm),
+  cardFormValidation: new FormValidator(validationObj, addCardForm),
+  picFormValidation: new FormValidator(validationObj, profilePicForm),
+};
 
 // initail setup
-cardPopup.setEventListeners();
-profilePopup.setEventListeners();
-locationPopup.setEventListeners();
+for (let popup in popups) {
+  popups[popup].setEventListeners();
+}
 
-cardSection.renderItems();
+for (let form in formValidators) {
+  formValidators[form].enableValidation();
+}
 
-editFormValidation.enableValidation();
-cardFormValidation.enableValidation();
+api
+  .getAll()
+  .then((data) => {
+    userInfo.setUserInfo(data[0]);
+    userInfo.setUserImage(data[0]);
+    return data;
+  })
+  .then((data) => {
+    popups.profilePopup.setInputs(userInfo.getUserInfo());
+    cardSection.renderItems(data[1]);
+  });
 
 // edits the user profile based off form inputs
 function editProfile(e, { name, description }) {
-  userInfo.setUserInfo({ name, description });
-  profilePopup.closeModal();
+  api
+    .updateUserInfo({ name, description })
+    .then(() => {
+      userInfo.setUserInfo({ name, description });
+      popups.profilePopup.closeModal();
+      popups.profilePopup.renderLoading();
```

Comment:
Needs correction:

1) renderLoading method has one argument
2) Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Fix that everywhere

Reply:
Since i use formValidators.editFormValidation.resetValidation() when i open the edit form (and resetValidation handles button state) should I only use the toggleButton on the other two input forms?

Reply:
I don't understand what you mean.. My comment is about stopping the loading in `finally` block, so the button text will return to it's initial state even if an error occurred.

Reply:
Sorry I misunderstood. I thought you meant its disabled/active state. Yes that makes sense about the loading the state.